homeassistant/components/switch/fritzbox.py

```diff
@@ -0,0 +1,109 @@
+"""
+Support for AVM Fritz!Box smarthome switch devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/switch.fritzbox/
+"""
+import logging
+
+import requests
+
+from homeassistant.components.fritzbox import DOMAIN as FRITZBOX_DOMAIN
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+DEPENDENCIES = ['fritzbox']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_CURRENT_CONSUMPTION = 'current_consumption'
+ATTR_CURRENT_CONSUMPTION_UNIT = 'current_consumption_unit'
+ATTR_CURRENT_CONSUMPTION_UNIT_VALUE = 'W'
+
+ATTR_TOTAL_CONSUMPTION = 'total_consumption'
+ATTR_TOTAL_CONSUMPTION_UNIT = 'total_consumption_unit'
+ATTR_TOTAL_CONSUMPTION_UNIT_VALUE = 'kWh'
+
+ATTR_TEMPERATURE_UNIT = 'temperature_unit'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Fritzbox smarthome switch platform."""
+    devices = []
+    fritz_list = hass.data[FRITZBOX_DOMAIN]
+
+    for fritz in fritz_list:
+        device_list = fritz.get_devices()
+        for device in device_list:
+            if device.has_switch:
+                devices.append(FritzboxSwitch(hass, device, fritz))
+
+    add_devices(devices)
+
+
+class FritzboxSwitch(SwitchDevice):
+    """The switch class for Fritzbox switches."""
+
+    def __init__(self, hass, device, fritz):
+        """Initialize the switch."""
+        self.units = hass.config.units
```

Comment:
The entity will have access to `hass` when it has been added to home assistant, so then you can use `self.hass.config.units`, so you can remove this attribute here.